pokemon/pokemon.py

```diff
@@ -0,0 +1,94 @@
+"""This module contains a two classes for creating pokemon."""
+
+from random import randint
+
+
+def repeat(m):
+    """Decorator-function allows decorated function to repeat m times."""
+
+    def inner(func_object):
+        def wrapper(*args, **kwargs):
+            print("Repeat ", m, " times")
+            for i in range(m):
+                func_object(*args, **kwargs)
+
+        return wrapper
+
+    return inner
+
+
+class Pokemon(object):
+    """
+    This class Pokemon represents a pokemon character with several attributes.
+
+     :param name: Name of pokemon
+     :type name: str
+     :param nickname: Nickname of pokemon
+     :type nickname: str
+     :param moves: list of 4 moves
+     :type moves: list
+    """
+
+    repeat_max = 10
+
+    def __init__(self, name, nickname):
+        """Initialise Pokemon class."""
+        self.name = name
+        self.nickname = nickname
+        self.moves = ["jump", "strike", "dash", "block"]
+
+    @repeat(randint(0, repeat_max))
+    def speak(self):
+        """Pokemon say name and repeat random times."""
+        print(self.name)
```

Comment:
So you've got the general idea here, but what will happen with this is the number will be chosen randomly once, and then forever more you'll be repeating the same number of times.

I'm not sure whether it'll be set for the class as a whole or each instance of the class, I'd need to sit and think about the inner workings, but I think it's the former.

What I had in mind was to repeat the function (i.e. printing the message) a different number of times, each time the `speak()` function is called.

Have a look at the code that I came up with earlier, and see where I use the `randint()` method. Try to see if you can understand why the difference is there:

```
from random import randint

def repeat_randomly(n=5):
  def decorator_repeat_randomly(func):
    def wrapper(*args, **kwargs):
      for _ in range(randint(1,n)):
        value = func(*args, **kwargs)
      return value
    return wrapper
  return decorator_repeat_randomly

@repeat_randomly(6)
def print_a_message(message):
  print(message)
```